app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -111,4 +112,21 @@ class MainActivity : AppCompatActivity() {
                 .replace(R.id.frameContainer, fragment)
                 .commit()
     }
+
+    override fun onBackPressed() {
+        val currentFragmentClass = this.supportFragmentManager.findFragmentById(R.id.frameContainer).javaClass
+        when (currentFragmentClass) {
+            SearchFragment().javaClass -> setEventFragment()
+            FavoriteFragment().javaClass -> setEventFragment()
+            OrdersUnderUserFragment().javaClass -> setEventFragment()
+            ProfileFragment().javaClass -> setEventFragment()
```

Comment:
why not just do it when currentFragmentClass is not EventFragment, if in future we want some different functionality we can change it

Reply:
> why not just do it when currentFragmentClass is not EventFragment, if in future we want some different functionality we can change it

Updated.